### PR TITLE
feat(cli): add --prebuilt flag to deploy command

### PR DESCRIPTION
**Pull request by matthewvolk**
## What/Why?

Adds a `--prebuilt` flag to `catalyst deploy` so users who have already run `catalyst build` separately can skip the automatic build step. When `--prebuilt` is passed but `.bigcommerce/dist/` is missing or empty, the CLI fails with an actionable error message directing the user to either run `catalyst build` first or remove the flag.

**Jira:** [CATALYST-1753](https://bigcommercecloud.atlassian.net/browse/CATALYST-1753)

### Changes
- Add `--prebuilt` option to the deploy command definition
- Replace unconditional `buildCatalystProject()` call with conditional logic that validates existing build output when `--prebuilt` is set
- Add tests: command instance check, build skip verification, missing dist failure, empty dist failure

## Testing

```bash
pnpm --filter @bigcommerce/catalyst run test   # 47 tests pass
pnpm --filter @bigcommerce/catalyst run lint    # 0 warnings
```

## Migration

N/A — additive change, no breaking behavior.

[CATALYST-1753]: https://bigcommercecloud.atlassian.net/browse/CATALYST-1753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jorgemoya — APPROVED** [review]

File: packages/catalyst/src/cli/commands/deploy.ts [hunk 1]
```diff
@@ -330,7 +333,30 @@ export const deploy = new Command('deploy')
...
-      await buildCatalystProject(projectUuid);
+      if (options.prebuilt) {
+        const distDir = join(process.cwd(), '.bigcommerce', 'dist');
+
+        try {
+          await access(distDir);
+        } catch {
+          throw new Error(
+            'No build output found at .bigcommerce/dist/. Run `catalyst build` first or remove `--prebuilt` to build automatically.',
+          );
```
Comment (migueloller) [inline packages/catalyst/src/cli/commands/deploy.ts:393]: I'm guessing we (or Commander.js) already handle thrown errors gracefully by logging the message and exiting?

**Review by migueloller — APPROVED** [review]
LGTM 👍🏻

**Review by chanceaclark — APPROVED** [review]

File: packages/catalyst/src/cli/commands/deploy.ts (hunk 1 above)
Comment (matthewvolk) [reply packages/catalyst/src/cli/commands/deploy.ts:393]: Correct - logged neatly using Consola: https://github.com/bigcommerce/catalyst/blob/fe96343ba019c2e08a7a7d976cff554fc4bfaa49/packages/catalyst/src/cli/commands/deploy.ts#L446-L449
